armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -429,10 +444,31 @@ class XSModelingOptions:
         (e.g., fission products) as a depletion product of an isotope with a much
         smaller atomic number.
 
+    averageByComponent: bool
```

Comment:
And the new settings just have to go in ARMI, because they're so closely tied to the current ARMI XSGM?

Reply:
`averageByComponent` was already in ARMI, it was just missing from the docstring.

The new settings `ductHeterogeneous` and `traceIsotopeThreshold` are here because they are new `XSModelingOptions`. There is no downstream extension of `XSModelingOptions`, it's all handled at this level.

`ductHeterogeneous` is also used as an argument passed to `blockConverters.HexComponentsToCylConverter`.